### ISSUE TITLE
0.2.2 in Safari chokes on window.BlobBuilder.prototype

### ISSUE THREAD

**Issue by robert-stuttaford** [shoreleave/shoreleave-browser#2]
In Safari and Opera (but not in Chrome or Firefox), i get this error:

``` javascript
window.BlobBuilder = shoreleave.browser.blob.blobber.call(null);
shoreleave.browser.blob.blob_builder = function blob_builder() {
  return new window.BlobBuilder
};
// here:
window.BlobBuilder.prototype.cljs$core$ITransientCollection$ = true;
```

The exception I get is: `'undefined' is not an object (evaluating 'window.BlobBuilder.prototype')`.

I am not using blobs in my code, so this is a whitespace-compilation-only issue. It means that I can not debug in Safari or Opera.

Please fix this :-)

For context, here are the cljs portions of my deps tree (using leiningen 2 preview10):

```
$ lein deps :tree
 [enfocus "1.0.0-alpha3"]
   [domina "1.0.0" :exclusions [[org.clojure/clojurescript]]]
     [goog-jar "1.0.0"]
 [shoreleave/shoreleave-pubsub "0.2.2"]
   [shoreleave/shoreleave-browser "0.2.2"]
   [shoreleave/shoreleave-core "0.2.2"]
   [shoreleave/shoreleave-worker "0.2.2"]
 [shoreleave/shoreleave-remote "0.2.2"]
```

**Comment by robert-stuttaford** [comment]
Update

I managed to dig goog-jar out after looking into domina. The problem still persists with the following deps:

```
 [domina "1.0.1-20120809.181807-1"] ;; SNAPSHOT
 [enfocus "1.0.0-alpha3" :exclusions [[domina] [org.clojure/clojurescript]]]
 [org.clojure/clojurescript "0.0-1450"]
   [com.google.javascript/closure-compiler "r1918"]
     [args4j "2.0.12"]
     [com.google.code.findbugs/jsr305 "1.3.9"]
     [com.google.guava/guava "10.0.1"]
     [com.google.protobuf/protobuf-java "2.4.1"]
     [com.googlecode.jarjar/jarjar "1.1"]
     [junit "4.8.2"]
     [org.apache.ant/ant "1.8.2"]
       [org.apache.ant/ant-launcher "1.8.2"]
     [org.json/json "20090211"]
   [org.mozilla/rhino "1.7R3"]
 [org.clojure/google-closure-library-third-party "0.0-1376-2"]
 [org.clojure/google-closure-library "0.0-1376-2"]
 [shoreleave/shoreleave-pubsub "0.2.2"]
   [shoreleave/shoreleave-browser "0.2.2"]
   [shoreleave/shoreleave-core "0.2.2"]
   [shoreleave/shoreleave-worker "0.2.2"]
 [shoreleave/shoreleave-remote "0.2.2"]
```

**Comment by ohpauleez** [comment]
The issue is happening because PubSub pulls in the worker, and the worker pulls in the offending Blob code.  I do know of Blob only working to spec on Chrome and Firefox.

Let me look at refactoring and reorganizing the code a bit to stop this from happening.
I'll also look into updates to the Blob spec.

**Comment by ohpauleez** [comment]
What version of Safari are you using, and does it have support for the Blob constructor?
https://developer.mozilla.org/en-US/docs/DOM/Blob#Browser_compatibility

**Comment by robert-stuttaford** [comment]
6.0 7536.25, on OSX Lion. According to that table, it should work.

**Comment by robert-stuttaford** [comment]
Thanks for your prompt response, Paul :-)

**Comment by ohpauleez** [comment]
This should be fixes in 0.2.3-SNAPSHOT
I haven't tested this at all, so let me know.  If it works for you, go ahead and close the ticket.  If not, I'll keep at it.

Also, you should always be debugging in Chrome and Firefox ;-)

**Comment by robert-stuttaford** [comment]
(Edited: removed idiocy)

Thanks Paul. That's working in Safari and Opera now.

Browsers: I am, I am! But I also need to spot-check in the other, lesser browsers -grin-
